app/models/reservation.rb

```diff
@@ -0,0 +1,22 @@
+class Reservation < ActiveRecord::Base
+  belongs_to :request
+  has_one :donor
+
+  state_machine :initial => :unreserved do
+    event :reserve do
+      transition :unreserved => :reserved
+    end
+
+    event :ship do
+      transition :reserved => :shipped
+    end
+
+    event :receive do
+      transition :shipped => :received
+    end
+
+    event :cancel do
+      transition :reserved => :unreserved, :shipped => :reserved
```

Comment:
I think I was so sucked into developer mode when writing that portion that I failed to remember that a reservation only exists if it's at the very least in a reserved state (and nothing less)